flink/interpreter/src/main/java/org/apache/zeppelin/flink/JobManager.java

```diff
@@ -254,11 +255,12 @@ public void run() {
               if (completedObject.has("external_path")) {
                 String checkpointPath = completedObject.getString("external_path");
                 LOGGER.debug("Latest checkpoint path: {}", checkpointPath);
-                if (!StringUtils.isBlank(checkpointPath)) {
+                if (!StringUtils.isBlank(checkpointPath) && !checkpointPath.equals(latestCheckpointPath)) {
```

Comment:
It seems that using the operator "not (!)" only once is good for readability.
```suggestion
                if (!(StringUtils.isBlank(checkpointPath) || checkpointPath.equals(latestCheckpointPath))) {
```

Reply:
Actually here 2 not is more readable. The if condition is when checkpoint is not null and it is not the same as latestCheckpointPath. 
If I change it to 1 not, it is a little harder to understand what the if condition means